course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -128,19 +160,540 @@ describe('Unit testing PeopleController', function() {
+            });
+        });
+
+        describe('handleAjaxError', function() {
+            it('should log an error', function() {
+                var expectedMessage = 
+                    "Error getting data from https://tea.pot: 418 I'm a teapot: {}";
+                scope.handleAjaxError({}, 418, {}, {url: 'https://tea.pot'},
+                                      "I'm a teapot");
+                expect($log.error.logs).toEqual([[expectedMessage]]);
+            });
+        });
+
+        describe('isEmailAddress', function() {
+            // TODO - steal test addresses from real email parsing library?
+            // all valid addresses, should return true
+            [{description: 'simple email address', testString: 'bob_dobbs@harvard.edu'},
+             {description: 'email address with long domain',
+              testString: 'bob_dobbs@very.specific.server.harvard.edu'},
+            ].forEach(function(test) {
+                it('should return true for ' + test.description, function() {
+                    expect(scope.isEmailAddress(test.testString)).toBe(true);
+                });
+            });
+
+            // not addresses, should return false
+            [{description: 'series of digits', testString: '123456789'},
+             {description: '@ but no .', testString: 'bob_dobbs@harvard'},
+             {description: '. but no @', testString: 'bob_dobbs.harvard.edu'},
+            ].forEach(function(test) {
+                it('should return false for ' + test.description, function() {
+                    expect(scope.isEmailAddress(test.testString)).toBe(false);
+                });
+            });
+        });
+
+        describe('selectRole', function() {
+            it('should set the role on the scope', function() {
+                var role = {roleId: 123};
+                scope.selectRole(role);
+                expect(scope.selectedRole).toEqual(role);
+            });
+        });
+    });
+
+    describe('addUser', function() {
+        beforeEach(function() {
+            var ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'addUser tests',
+            };
+            courseInstances.instances[ci.course_instance_id] = ci;
+            controller = $controller('PeopleController', {$scope: scope});
+            spyOn(scope, 'addUserToCourse');
+            spyOn(scope, 'lookup');
+        });
+
+        it('should call lookup if there are no search results', function() {
+            scope.searchResults = [];
+            scope.addUser('bob');
+            expect(scope.addUserToCourse.calls.count()).toEqual(0);
+            expect(scope.lookup.calls.count()).toEqual(1);
+            expect(scope.lookup.calls.argsFor(0)).toEqual(['bob']);
+        });
+
+        it('should log an error if called with a single search result', function() {
+            scope.searchResults = [{}]; // contents don't matter
```

Comment:
(referencing above comment re: making test data more 'real-world'-like, here, 'single search result' is represented by `[{}]` (because the number actually matters when calling `addUser()`, whereas above it's conflated with multiple results.)